src/modules/navigator/mission.cpp

```diff
@@ -270,6 +263,13 @@ Mission::on_active()
 
 		do_abort_landing();
 	}
+
+	if (_work_item_type == WORK_ITEM_TYPE_PRECISION_LAND) {
+		_navigator->get_precland()->on_active();
+
+	} else if (_navigator->get_precland()->is_activated()) {
+		_navigator->get_precland()->on_inactivation();
```

Comment:
I don't understand why we have to do that because if we're in `Mission` already we are not in `PrecLand` and `precland->on_inactive()` should have been called.

Reply:
We could be in mission and precland. Calling precland as a landing part of the mission was already supported, I just added the same possibility for rtl. Also made the mission-precland calling a bit more readable, and enabled it to be called at the same iteration when it switches by moving it from the begining to the end of mission::on_active

Reply:
I agree it looks hacky calling one nav mode from another, also its been like that for a long time.

IMO the proper way to handle precision landing is moving it to a (separate) flight task, which is also a bigger project. That way we could have different velocity profile and and act more aggressively since we are dealing with position errors of centimeters to which the controller almost doesnt respond. Currently everything is tuned to work smoothly when following waypoints where position is controlled on a meters scale. 

To get the drone land precisely on a becon thats only 10cm wide we had to hack in another controller in precland to exaggerate the errors in order to make the control more aggressive and handle these small errors.

Reply:
Ok, I guess you have studied and tested this enough to be sure that it's a sensible step for now.
The thing with the flight task sounds sensible.

Reply:
As far as the testing goes, we are using these changes since January, and end every mission with precision landing, so the changes in mission.cpp were used on more than a 100 flights I'd say.

Changes in RTL were tested on a few flights, both simulation and real world, since we dont usually go into RTL mode often. Tests were successful.
Although the added logic is simple, any other opinions, or suggestion for corner cases tests are appreciated.